build/stf-run-ci/defaults/main.yml

```diff
@@ -21,3 +21,11 @@ sg_core_image_tag: latest
 sg_bridge_image_tag: latest
 prometheus_webhook_snmp_image_tag: latest
 namespace: service-telemetry
+
+# used when building images to default to correct version branch for STF subcomponents per STF version
+version_branches:
+  sgo: stable-1.3
+  sg_core: stable-1.3
+  sg_bridge: stable-1.2
+  prometheus_webhook_snmp: stable-1.2
```

Comment:
Hmm, the comment said "Considering not all components always follow the 'stable-' convention..." but it looks like they DO follow that convention. It seems the reason this code needs to exist (right now) is simply that we neglected to create stable-1.3 branches in some of the dependencies.  I still like this concept and think the implementation is valuable, I just feel slightly misled ;)

Reply:
Yea since prometheus-webhook-snmp (and sg-bridge) didn't change for the STF 1.3 release, corresponding branches were not created. Ultimately the branch naming is probably a bit of a mislead overall since I think the intention was that we'd have per y-stream release component branches, but it appears that's not always the case.

In our downstream productization stream for example, we have an `stf-common` pipeline that takes in those common release components across STF 1.2 and 1.3, which relates to the `stable-1.2` branches currently upstream.

We could look at adjusting this, but basically some components are used across multiple product releases under a single version. I'm not exactly sure how to resolve this other than to duplicate the branching to be consistent, but that feels more confusing and managed overhead than necessary.

For the logging work for example, we'd in theory have a `stable-1.2` and `stable-1.4` branch for `sg-bridge` with no `stable-1.3` release. Or maybe it'd become `stable-2.0` and be used as part of STF 1.4.

Reply:
Yeah I think markdown removed the text after "stable-" in my comment, but it was supposed to indicate that the stable version for the sub component may not always match the STF version. Sorry for the confusion